govc/flags/client.go

```diff
@@ -428,11 +428,15 @@ func (flag *ClientFlag) newClient() (*vim25.Client, error) {
 	m := session.NewManager(c)
 	u := flag.url.User
 
-	if u.Username() == "" {
-		// Assume we are running on an ESX or Workstation host if no username is provided
-		u, err = flag.localTicket(ctx, m)
-		if err != nil {
-			return nil, err
+	if u.Username() == "" && !c.IsVC() {
+		// If no username is provided, try to acquire a local ticket.
+		// When invoked remotely, ESX returns an InvalidRequestFault.
+		// So, rather than return an error here, fallthrough to Login() with the original User to
+		// to avoid what would be a confusing error message.
```

Comment:
so error happens here, we login with empty user, and get something like `LoginFault`, which is less confusing. Is that accurate?

Reply:
exactly, InvalidLoginFault with the appropriate localized message.